src/gt4py/next/program_processors/runners/dace_fieldview/gtir_builtin_translators.py

```diff
@@ -121,21 +117,36 @@ def _parse_fieldop_arg(
     if not isinstance(arg, FieldopData):
         raise ValueError(f"Received {node} as argument to field operator, expected a field.")
 
-    if isinstance(arg.gt_dtype, ts.ScalarType):
-        return gtir_dataflow.MemletExpr(arg.dc_node, sbs.Indices([0]))
-    elif isinstance(arg.gt_dtype, ts.FieldType):
+    if isinstance(arg.gt_type, ts.ScalarType):
+        return gtir_dataflow.MemletExpr(
+            dc_node=arg.dc_node, gt_dtype=arg.gt_type, subset=sbs.Indices([0])
+        )
+    elif isinstance(arg.gt_type, ts.FieldType):
+        field_dtype: itir_ts.ListType | ts.ScalarType
```

Comment:
I don't like uninitialized variables, maybe extract a function to get this information.

Reply:
Fixed.